### PR TITLE
Add ErrorMessage Override

### PR DESCRIPTION
**Pull request by LinnJS**
## Description
As a user I would like to see more than a hard coded error for every form in the FormBuilder, please make an enhancement so we can override the default hard coded ErrorMessage with our own Custom ErrrorMessage alert.

[ZenHub issue](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/10233)

Add a ternary so this default error can be replaced by a custom DSVA Error alert component

That custom though...
Custom Paint

## Screenshots
Screen Shot 2020-06-17 at 11 07 21 AM

## Acceptance criteria
- [x] Accepts component as a prop, and overrides the default ErrorMessage

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by LinnJS** [comment]
Having this issue whenever trying to mount this component I updated... not sure how to test this functionality without a full mount...

Screen Shot 2020-06-29 at 5 20 44 PM

File: src/platform/forms/tests/save-in-progress/RoutedSavableReviewPage.unit.spec.jsx [hunk 1]
```diff
@@ -227,7 +227,7 @@ describe('Schemaform save in progress: RoutedSavableReviewPage', () => {
 
     let tree;
     beforeEach(() => {
-      tree = shallow(
+      tree = mount(
```
Comment (LinnJS) [inline src/platform/forms/tests/save-in-progress/RoutedSavableReviewPage.unit.spec.jsx:230]: how can I go about testing this feature without a full mount? Not sure how to check to see if children are present without a full mount...

**Comment by LinnJS** [comment]
Still stuck on this error, not sure how to continue, seems like there are pre-existing bugs with the component I am trying to add test-cases to

image

**Comment by LinnJS** [comment]
Not required to do unit testing due to pre-existing VSP error as the conversation in the attack ticket alludes. Here is a screen shot of the Q&A.

Screen Shot 2020-07-13 at 3 49 12 PM

**Review by bradl3yC — APPROVED** [review]

File: src/platform/forms/save-in-progress/RoutedSavableReviewPage.jsx [hunk 2]
```diff
@@ -82,38 +83,47 @@ class RoutedSavableReviewPage extends React.Component {
...
             <p>
-              If you don’t have an account, you’ll have to start over. Try
-              submitting your form again tomorrow.
+              We’re working to fix the problem. Please make sure you’re
+              connected to the Internet, and then try saving your form again.{' '}
+              {saveLink}.
             </p>
-          )}
-          <InlineErrorComponent />
+            {!user.login.currentlyLoggedIn && (
```
Comment (mpelzsherman) [inline src/platform/forms/save-in-progress/RoutedSavableReviewPage.jsx:111]: Are we concerned about Law of Demeter issues here? Will user.login always be present with data type of object?

File: src/platform/forms/save-in-progress/RoutedSavableReviewPage.jsx [hunk 3]
```diff
@@ -82,38 +83,47 @@ class RoutedSavableReviewPage extends React.Component {
...
+          </div>
         </div>
-      </div>
+      );
+    };
+
+    return CustomSubmissionError ? (
+      <CustomSubmissionError location={location} form={form} user={user} />
+    ) : (
+      <DefaultErrorMessage />
```
Comment (mpelzsherman) [inline src/platform/forms/save-in-progress/RoutedSavableReviewPage.jsx:126]: Shouldn't we add a test for this?

I'd also be interested in seeing an example of adding a CustomSubmissionError in the form config.

Comment (LinnJS) [reply src/platform/forms/save-in-progress/RoutedSavableReviewPage.jsx:126]: We cannot test this due to not being able to do a full mount in this component to see the children due to pre-existing tech debt as stated in the above comment.

As for examples, I have another PR with a custom error component implemented as well as updated documentation with an example.

Implementation PR: 
https://github.com/department-of-veterans-affairs/vets-website/pull/13298

Documentation PR:
https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/307

File: src/platform/forms/save-in-progress/RoutedSavableReviewPage.jsx (hunk 2 above)
Comment (LinnJS) [reply src/platform/forms/save-in-progress/RoutedSavableReviewPage.jsx:111]: I honestly am not sure to be honest with you, this was code that was already there that I just moved into a function. Perhaps when a VSP member can answer this question when they review this PR. If it is not always there more than willing to throw some `optional chaining` in there.

Comment (mpelzsherman) [reply src/platform/forms/save-in-progress/RoutedSavableReviewPage.jsx:111]: I'm ok leaving it as it. There are tons of places in the code where we call `user.login.currentlyLoggedIn` with impunity, so it must be safe to do so.

File: src/platform/forms/save-in-progress/RoutedSavableReviewPage.jsx (hunk 3 above)
Comment (mpelzsherman) [reply src/platform/forms/save-in-progress/RoutedSavableReviewPage.jsx:126]: OK thanks @LinnJS .

**Review by mpelzsherman — APPROVED** [review]

**Review by Clark-W-Griswold — APPROVED** [review]
👍

**Pull request merged**